pages/apim/3.x/installation-guide/installation-guide-plugins.adoc

```diff
@@ -0,0 +1,187 @@
+= Installing and updating Plugins
+:page-sidebar: apim_3_x_sidebar
+:page-permalink: apim/3.x/apim_installation_guide_plugins.html
+:page-folder: apim/overview
+:page-layout: apim3x
+:page-liquid:
+:page-description: Gravitee.io API Management - Installation Guide - Plugins
+:page-keywords: Gravitee.io, API Management, apim, guide, installation, upgrade, plugins
+:page-toc: true
+
+// author: Tom Geudens
+== Overview
+Plugins bring Gravitee APIM to life. They provide functionality ranging from connectors (eg. Kafka) to tracers (eg. Jaeger) and everything in between. A special mention goes to policies, plugins that provide you with options to shape and control the flow of your APIs.
+
+A lot of plugins are part of the Gravitee APIM installation. Ones that aren't generally applicable can be added. You can even write your own! This document explains how you can install and upgrade additional plugins.
+
+== In a nutshell - Option I
+. Get your plugin zip-file ready.
++
+The supported plugins for Gravitee APIM can be found link:https://download.gravitee.io/#graviteeio-apim/plugins/[here, window=\"_blank\"].
+
+. Drop the plugin zip-file in the plugins-folder of the relevant component. This could be the APIM REST API, the APIM Gateway or both!
+
+. Restart the component.
+
+NOTE: At this time installing and updating plugins requires a restart of the component.
+
+== In a nutshell - Option II
+. Get your plugin zip-file ready.
++
+The supported plugins for Gravitee APIM can be found link:https://download.gravitee.io/#graviteeio-apim/plugins/[here, window=\"_blank\"].
+
+. Drop the plugin zip-file in a plugins-folder of your own choosing.
+
+. Modify the `gravitee.yml` configuration file of the relevant component (APIM REST API / APIM Gateway / both) to take the additional folder into account
+
+. Restart the component.
+
+NOTE: At this time installing and updating plugins requires a restart of the component.
+
+== Details for .ZIP installations
+=== Option I
+. Locate the plugins folder of the component
++
+*  APIM REST API - <your installation folder>/graviteeio-apim-rest-api-{{ site.products.apim._3x.version }}/plugins
+*  APIM Gateway - <your installation folder>/graviteeio-apim-gateway-{{ site.products.apim._3x.version }}/plugins
+
+. Drop the plugin zip file into the folder
+
+. Restart the component
+
+=== Option II
+. Create a custom location for plugins
++
+[source,bash]
+----
+mkdir /customlocation/plugins
+----
++
+NOTE: This is an example location, yours can differ.
+
+. Drop the plugin zip-file into that folder
+
+. Modify the `gravitee.yml` configuration file of the component
++
+*  APIM REST API - <your installation folder>/graviteeio-apim-rest-api-{{ site.products.apim._3x.version }}/config
+*  APIM Gateway - <your installation folder>/graviteeio-apim-gateway-{{ site.products.apim._3x.version }}/config
+
++
+[source,yaml]
+----
+# Plugins repository
+plugins:
+  path:
+    - ${gravitee.home}/plugins
+    - /customlocation/plugins
```

Comment:
Global reflexion about 'customLocation'.

A confused user could set the same plugin custom location on both rest-api and gateway side.
And will get errors running a component with plugins dedicated to another.

Maybe we have to explicit that there are 2 different folders :
- 1 for rest-api plugins
- 1 for gateway plugins